File: examples/nodejs/subscribe_quote.js
const { Config, QuoteContext, SubType, OAuth, OAuthToken } = require("longport");

let globalCtx;

async function getToken() {
  const clientId = "your-client-id";
  try {
    const token = OAuthToken.load();
    if (token.isExpired()) throw new Error("token expired");
    if (token.expiresSoon()) {
      const oauth = new OAuth(clientId);
      try {
        const newToken = await oauth.refresh(token);
        newToken.save();
        return newToken;
      } catch (_) {}
    } else {
      return token;
    }
  } catch (_) {}
  const oauth = new OAuth(clientId);
  const token = await oauth.authorize((url) => {
    console.log("Open this URL to authorize: " + url);
  });
  token.save();
  return token;
}

async function main() {
  const token = await getToken();
  let config = Config.fromOAuth(token);
  globalCtx = await QuoteContext.new(config);
  globalCtx.setOnQuote((_, event) => console.log(event.toString()));
  globalCtx.subscribe(["TSLA.US"], [SubType.Quote], true);
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/subscribe_quote.py
import asyncio
from time import sleep

from longport.openapi import QuoteContext, Config, OAuth, OAuthToken, SubType, PushQuote


def on_quote(symbol: str, event: PushQuote):
    print(symbol, event)


async def get_config() -> Config:
    try:
        token = OAuthToken.load()
    except Exception:
        oauth = OAuth("your-client-id")
        token = await oauth.authorize(lambda url: print(f"Open this URL to authorize: {url}"))
        token.save()
    return Config.from_oauth(token)


config = asyncio.run(get_config())
ctx = QuoteContext(config)
ctx.set_on_quote(on_quote)
ctx.subscribe(
    ["700.HK", "AAPL.US", "TSLA.US", "NFLX.US"],
    [SubType.Quote],
    is_first_push=True,
)
sleep(10)
